.github/workflows/lint.yml

```diff
@@ -2,7 +2,11 @@ name: Linting and style checking
 
 on:
   push:
+    branches:
+      - "master"
```

Comment:
I'm a bit confused why this is needed for push on master. Isn't linting only useful before merging a PR?

Reply:
It was there when I found it... I'm not sure why "push" is needed at all? (Assuming you don't push straight to master -- which sometimes happens...)

Reply:
Even in that case it's too late since it's already merged. I think you can remove the entire push section tbh.

Reply:
For both actions?

Reply:
Hmm, I don't think so. The `lint.yml` workflow doesn't need it but it looks like `update-readme.yml` needs it (since it uses a push to master to check if readme should be updated.)

Reply:
Gotcha; I was thinking of the tests workflows.

Reply:
Could be worth it since PRs can use the cache from the master branch workflow. Aka not guaranteed useless but haven't benchmarked or anything.